core/network/subnet_test.go

```diff
@@ -219,21 +219,31 @@ func (*subnetSuite) TestSubnetInfosGetByUnderLayCIDR(c *gc.C) {
 
 func (*subnetSuite) TestSubnetInfosGetByCIDR(c *gc.C) {
 	s := network.SubnetInfos{
-		{ID: "1", CIDR: "10.10.10.0/24", ProviderId: "1"},
-		{ID: "2", CIDR: "10.10.10.0/24", ProviderId: "2"},
-		{ID: "3", CIDR: "20.20.20.0/24"},
+		{ID: "1", CIDR: "10.10.10.0/25", ProviderId: "1"},
+		{ID: "2", CIDR: "10.10.10.0/25", ProviderId: "2"},
+		{ID: "3", CIDR: "20.20.20.0/25"},
 	}
 
 	_, err := s.GetByCIDR("invalid")
 	c.Check(err, jc.Satisfies, errors.IsNotValid)
 
-	subs, err := s.GetByCIDR("30.30.30.0/24")
+	subs, err := s.GetByCIDR("30.30.30.0/25")
 	c.Assert(err, jc.ErrorIsNil)
 	c.Check(subs, gc.HasLen, 0)
 
-	subs, err = s.GetByCIDR("10.10.10.0/24")
+	subs, err = s.GetByCIDR("10.10.10.0/25")
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(subs.EqualTo(s[:2]), jc.IsTrue)
+
+	// Check fallback CIDR-in-CIDR matching when CIDR is carved out of a
+	// subnet CIDR.
+	subs, err = s.GetByCIDR("10.10.10.0/31")
```

Comment:
I think we need a check for "10.10.10.2/31" which should also be in the same parent subnet (if I understand what they are doing)